opl/opl_sdl.c

```diff
@@ -173,8 +173,8 @@ static void FillBuffer(int16_t *buffer, unsigned int nsamples)
 
         for (i=0; i<nsamples; ++i)
         {
-            buffer[i * 2] = (int16_t) mix_buffer[i * 2];
-            buffer[i * 2 + 1] = (int16_t) mix_buffer[i * 2 + 1];
+            buffer[i * 2 + 1] = (int16_t) mix_buffer[i * 2];
```

Comment:
The swap that's taking place here implies a deeper problem - ie. the dbopl.c code is generating the two channels backwards. ```mix_buffer[i * 2]``` should be the left channel and ```mix_buffer[i * 2 + 1]``` the right channel. So it would be better to fix the problem at its source at its source.

A cursory examination of dbopl.c shows that the only real stereo effect [appears to be generated by the maskLeft and maskRight variables](https://github.com/chocolate-doom/chocolate-doom/blob/master/opl/dbopl.c#L827). Cross-referencing with the [YMF262 datasheet](http://map.grauw.nl/resources/sound/yamaha_ymf262.pdf) we see that the bitmasks are referring to the CHA and CHB fields which control output for different channels (page 5). So are we saying that CHA is right and CHB is left? If so, we should just fix that.